tests/Jobs/RockTheVote/ImportRockTheVoteRecordTest.php

```diff
@@ -533,211 +533,284 @@ public function testSmsSubscriptionPayloadIsEmptyIfUserAlreadyHasMobile()
     }
 
     /**
-     * ---------------------------
-     * parseSmsStatusChangeForUser
-     * ---------------------------
-     */
-
-    /**
-     * Return mock user and row for a parseSmsStatusChange test.
+     * Return mock user and row that do not have voter registration changes for a
+     * SmsSubscriptionUpdate test.
      *
      * @param string $currentSmsStatus
      * @param bool @rtvSmsOptIn
      * @return obj
      */
-    public function getMocksForParseSmsStatusChangeTest($currentSmsStatus, bool $rtvSmsOptIn)
+    public function getMocksForUpdateUserSmsTest($currentSmsStatus, bool $rtvSmsOptIn)
     {
         $user = new NorthstarUser([
             'id' => $this->faker->northstar_id,
             'mobile' => $this->faker->phoneNumber,
             'sms_status' => $currentSmsStatus,
+            'sms_subscription_topics' => in_array($currentSmsStatus, [SmsStatus::$active, SmsStatus::$less, SmsStatus::$pending]) ? ['general', 'voting'] : [],
+            'voter_registration_status' => 'registration_complete',
         ]);
+
         $row = $this->faker->rockTheVoteReportRow([
             RockTheVoteRecord::$mobileFieldName => $this->faker->phoneNumber,
             RockTheVoteRecord::$smsOptInFieldName => $rtvSmsOptIn ? 'Yes' : 'No',
+            'Status' => 'Complete',
+            'Finish with State' => 'Yes',
         ]);
 
         return (object) ['user' => $user, 'row' => $row];
     }
 
     /**
-     * Tests SMS status is updated to active if user with null value opts-in.
-     *
      * @return void
      */
-    public function testParseSmsStatusChangeIfUserHasNullSmsStatusAndOptsIn()
+    public function testEnabledUpdateUserSmsWhenUserHasNullSmsStatusAndOptsIn()
     {
-        $mocks = $this->getMocksForParseSmsStatusChangeTest(null, true);
+        $this->enableUpdateUserSmsFeature(true);
+
+        $mocks = $this->getMocksForUpdateUserSmsTest(null, true);
         $job = new ImportRockTheVoteRecord($mocks->row, factory(ImportFile::class)->create());
 
-        $result = $job->parseSmsStatusChangeForUser($mocks->user);
+        $this->northstarMock->shouldReceive('updateUser')
+          ->with($mocks->user->id, [
+              'sms_status' => SmsStatus::$active,
+              'sms_subscription_topics' => ['voting'],
+          ])
+          ->andReturn($mocks->user);
 
-        $this->assertEquals(['sms_status' => SmsStatus::$active], $result);
+        $job->updateUserIfChanged($mocks->user);
+
+        $this->enableUpdateUserSmsFeature(false);
```

Comment:
Each test should be a fresh application environment, so it shouldn't be necessary to reset this to `false`. (If that _isn't_ the case & subsequent tests inherit the same application state, then we should fix that in `TestCase`!)